cli/js/compiler.ts

```diff
@@ -1380,21 +1421,94 @@ function compile({
   };
 }
 
+function transpile({
+  config: configText,
+  configPath,
+  cwd,
+  performance,
+  sourceFiles,
+}: TranspileRequest): CompileResponse {
+  if (performance) {
+    performanceStart();
+  }
+  log(">>> transpile start");
+  const compilerOptions: ts.CompilerOptions = Object.assign(
+    {},
+    DEFAULT_TRANSPILE_OPTIONS
+  );
+  if (configText && configPath && cwd) {
+    const { config, error } = ts.parseConfigFileTextToJson(
+      configPath,
+      configText
+    );
+    if (error) {
+      return { diagnostics: fromTypeScriptDiagnostic([error]), emitMap: {} };
+    }
+    const { options, errors } = ts.convertCompilerOptionsFromJson(
+      config.compilerOptions,
+      cwd
+    );
+    if (errors) {
+      return { diagnostics: fromTypeScriptDiagnostic(errors), emitMap: {} };
+    }
+
+    const ignoredOptions: string[] = [];
+    for (const key of Object.keys(options)) {
+      if (
+        IGNORED_COMPILER_OPTIONS.includes(key) &&
+        (!(key in compilerOptions) || options[key] !== compilerOptions[key])
+      ) {
+        ignoredOptions.push(key);
+        delete options[key];
+      }
+    }
+    Object.assign(compilerOptions, options);
+  }
```

Comment:
Why not `compiler.configure()`?

Reply:
You mean `host.configure()` which there is no host in transpile.  We have this logic broken out of the host in `deno_typescript` so it makes sense to break it out here too and have the `host.configure()` call that as well.